File: 0724-find-pivot-index/0724-find-pivot-index.java
class Solution {
    public int pivotIndex(int[] nums) {
     ArrayList<Integer>left=new ArrayList<>();
     ArrayList<Integer>right=new ArrayList<>();
     int sum=0;
     for(int i=0;i<nums.length;i++)
     {
        left.add(sum);
        sum += nums[i];
     }
     sum=0;
     for(int i=nums.length-1;i>=0;i--)
     {
        right.add(sum);
        sum += nums[i];
     }
     Collections.reverse(right);
      for(int i=0;i<nums.length;i++)
      {
        if(left.get(i).equals(right.get(i)))
        {
            return i;
        }
      }
     return -1;

        
    }
}

File: 0724-find-pivot-index/0724-find-pivot-index.cpp
class Solution {
public:
    int pivotIndex(vector<int>& nums) {
        vector<int>L;
        vector<int>R;
        int Lsum=0;
        int Rsum=0;

        for(int i=0;i<nums.size();i++)
        {
            L.push_back(Lsum);
            Lsum+=nums[i];
          
        }
         for(int i=nums.size()-1;i>=0;i--)
        {
            R.push_back(Rsum);
            Rsum +=nums[i];
        }
         reverse(R.begin(), R.end());
        for(int i=0;i<nums.size();i++)
        {
           if(L[i]==R[i])
           {
            return i;
           }
        }
        return -1;
        
    }
};